test/sharness/t0053-dag.sh

```diff
@@ -81,25 +194,46 @@ test_dag_cmd() {
   '
 
   test_expect_success "sub-objects look right" '
-    echo "\"world\"" > sub1_exp &&
+    echo -n "\"world\"" > sub1_exp &&
     test_cmp sub1_exp sub1 &&
-    echo "{\"beep\":[0,\"bop\"],\"dict\":\"ionary\"}" > sub2_exp &&
+    echo -n "{\"beep\":[0,\"bop\"],\"dict\":\"ionary\"}" > sub2_exp &&
     test_cmp sub2_exp sub2 &&
-    echo "[0,\"bop\"]" > sub3_exp &&
+    echo -n "[0,\"bop\"]" > sub3_exp &&
     test_cmp sub3_exp sub3 &&
-    echo "0" > sub4_exp &&
+    echo -n "0" > sub4_exp &&
     test_cmp sub4_exp sub4 &&
-    echo "\"bop\"" > sub5_exp &&
+    echo -n "\"bop\"" > sub5_exp &&
     test_cmp sub5_exp sub5
   '
 
-  test_expect_success "can pin cbor object" '
-    ipfs pin add $EXPHASH
+  test_expect_success "traversals using /ipld/ work" '
+    ipfs dag get /ipld/$IPLDPBHASH/Links/0/Name > link0name_actual
+  '
+
+  test_expect_success "retrieved node looks right" '
+    echo -n "\"some link\"" > link0name_expected
+    test_cmp link0name_actual link0name_expected
+  '
+
+  test_expect_success "can pin ipld object" '
+    ipfs pin add $IPLDHASH
+  '
+
+  test_expect_success "can pin dag-pb object" '
+    ipfs pin add $IPLDPBHASH
+  '
+
+  test_expect_success "can pin dag-cbor object" '
+    ipfs pin add $IPLDCBORHASH
+  '
+
+  test_expect_success "can pin dag-json object" '
+    ipfs pin add $IPLDJSONHASH
   '
```

Comment:
@willscott @aschmahmann so ... I've done a bunch of cleanup in 0053 to make it make sense and also test some additional stuff (including the `/ipld/` pathing in dag-pb that's now been enabled). But pinning is now no longer working for these last 3, although the first one works and I have no explanation for it. Just above this pinning section you'll see the `dag get /ipld/$IPLDPBHASH/Links/0/Name` working correctly, yet when we `ipfs pin add $IPLDPBHASH` it doesn't work. But it's also not working for the dag-cbor or dag-json forms either but it is working for `$IPLDHASH` which is the `{"hello":"world"...}` object.

`$IPLDCBORHASH`, `$IPLDJSONHASH` and `$IPLDPBHASH` are now references to this data: https://github.com/ipld/codec-fixtures/tree/master/fixtures/dagpb_2link%2Bdata - in 0053 we're taking the same binary form in each of these codecs, feeding it in and asking for it to be stored as each of the other 3 codecs. So 3x3 `dag add`s and checking the resulting CIDs, and it's all good and we can even `dag get` them. But they won't `dag pin`... It's `merkledag: not found`.

Any thoughts on where to look?

Reply:
oh 🤦 I've introduced an incomplete dag, so pin isn't happy about not having the blocks referenced, my bad, now to fix but keep enough of the test functionality intact.